components/esp_modem/examples/pppos_client/main/pppos_client_main.c

```diff
@@ -201,8 +205,15 @@ void app_main(void)
     xEventGroupWaitBits(event_group, CONNECT_BIT, pdTRUE, pdTRUE, portMAX_DELAY);
     /* Config MQTT */
     esp_mqtt_client_config_t mqtt_config = {
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
+            .broker = {
+                .address.uri = BROKER_URL
+            },
+			//TODO !! Event Handler !
```

Comment:
Should use event handler registration instead:

https://github.com/espressif/esp-idf/blob/36f49f361c001b49c538364056bc5d2d04c6f321/examples/protocols/mqtt/tcp/main/app_main.c#L142